packages/page-accounts/src/modals/LocalAdd.tsx

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017-2024 @polkadot/app-accounts authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Blockchain } from '@acala-network/chopsticks-core';
+import type { ActionStatus } from '@polkadot/react-components/Status/types';
+import type { ModalProps } from '../types.js';
+
+import { setStorage } from '@acala-network/chopsticks-core';
+import React, { useCallback, useState } from 'react';
+import store from 'store';
+
+import { Button, Input, InputAddressSimple, Modal } from '@polkadot/react-components';
+import { useApi } from '@polkadot/react-hooks';
+import { keyring } from '@polkadot/ui-keyring';
+
+import { useTranslation } from '../translate.js';
+
+interface Props extends ModalProps {
+  className?: string;
+  onClose: () => void;
+  onStatusChange: (status: ActionStatus) => void;
+}
+
+interface CreateOptions {
+  name: string;
+  tags?: string[];
+  balance?: number;
+}
+
+export async function createLocalAccount (chain: Blockchain, address: string, { balance = 1000, name }: CreateOptions): Promise<ActionStatus> {
+  const status = { action: 'create' } as ActionStatus;
+  const existing = store.get('localAccounts') as { address: string, name: string, balance: number }[] | undefined;
+
+  try {
+    await setStorage(chain, {
+      System: {
+        Account: [
+          ...(existing || []).map((account) => {
+            return [[account.address], { data: { free: account.balance * 1e12 }, providers: 1 }];
```

Comment:
I think we should not change user state. This account should remain as is. If founds are needed then user can use Alice account to fund the address